specification/machinelearningservices/AzureAI.Assets/routes.tsp

```diff
@@ -120,3 +134,118 @@ interface Indexes {
     PagedIndex
   >;
 }
+
+@added(Versions.`2024-05-01-preview`)
+interface Prompts {
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Waiting for bug fix: https://github.com/Azure/typespec-azure-pr/issues/3739."
+  @doc("Get a specific version of a Prompt.")
+  @route("/prompts/{name}/versions/{version}")
+  @get
+  get is Azure.Core.Foundations.Operation<
+    {
+      @doc("Name of the prompt")
+      @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_]*$")
+      @maxLength(254)
+      @path
+      name: string;
+
+      @doc("Version of the prompt")
+      @path
+      version: string;
+    },
+    Prompt
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Waiting for bug fix: https://github.com/Azure/typespec-azure-pr/issues/3739."
+  @doc("Creates or updates a Prompt")
+  @route("/prompts/{name}/versions/{version}")
+  @put
+  createOrUpdate is Azure.Core.Foundations.Operation<
+    {
+      @doc("Name of the prompt")
+      @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_]*$")
+      @maxLength(254)
+      @path
+      name: string;
+
+      @doc("Version of the prompt")
+      @path
+      version: string;
+
+      @doc("Properties of a Prompt Version.")
+      @body
+      body: Prompt;
+    },
+    ResourceCreatedOrOkResponse<Prompt>
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Waiting for bug fix: https://github.com/Azure/typespec-azure-pr/issues/3739."
+  @doc("List the versions of a Prompt given the name.")
+  @route("/prompts/{name}/versions")
+  @get
+  list is Azure.Core.Foundations.Operation<
+    {
+      @doc("Name of the prompt")
+      @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_]*$")
+      @maxLength(254)
+      @path
+      name: string;
+
+      @doc("View type for including/excluding (for example) archived entities.")
+      @query
+      listViewType: string = "ActiveOnly";
+
+      @doc("Ordering of list: Please choose orderby value from ['createdAt', 'lastModifiedAt'].")
+      @query
+      orderby?: string;
+
+      @doc("Comma-separated list of tag names (and optionally values). Example: tag1,tag2=value2.")
+      @query
+      tags?: string;
```

Comment:
tsp to define CSV, e.g.
```ts
  @query({
    format: "csv",
  })
  tags?: string[];
```

Reply:
so it's a string, not an array of strings

Reply:
`@query` with `format` basically describe in typepec syntax, what you are currently describing in the doc.